商品詳細表示機能

takanori-hagiwara opened the pull request:
# WHAT
FURIMA app商品詳細機能の実装
# WHY
GYAZO
- ログイン状態の出品者が、自身の出品した販売中商品の詳細ページへ遷移した動画
https://gyazo.com/ef0ddb983c9e2e00ac85fb0b3433a9d1
- ログイン状態の出品者以外のユーザーが、他者の出品した販売中商品の詳細ページへ遷移した動画
https://gyazo.com/62466d33644395143114dc5828c234dc
- ログアウト状態のユーザーが、商品詳細ページへ遷移した動画
https://gyazo.com/48d67a1bacb5a3f6ef8984b0151da7c4

takanori-hagiwara committed:
商品詳細表示機能

tatsuki-kato commented on app/controllers/items_controller.rb:
```diff
@@ -1,5 +1,5 @@
 class ItemsController < ApplicationController
-  before_action :authenticate_user!, only: [:new, :create]
+  before_action :authenticate_user!, only: [:new, :create, :show]
```
```rb
before_action :authenticate_user!, only: [:new, :create, :show]
```
こちらの中のshowを削除しましょう！
上記にshowアクションを追加してしまうと、
```
- ログアウト状態のユーザーでも、商品詳細表示ページを閲覧できること
```
が実装できなくなってしまうためです。

tatsuki-kato commented on app/views/items/show.html.erb:
```diff
@@ -43,27 +45,27 @@
+          <td class="detail-value"><%= @item.user.name %></td>
         </tr>
         <tr>
           <th class="detail-item">カテゴリー</th>
-          <td class="detail-value"><%= "カテゴリー名" %></td>
+          <td class="detail-value"><%= @item.category.name %></td>
         </tr>
```
```erb
  <a href="#" class="another-item"><%= "商品のカテゴリー名" %>をもっと見る</a>
```
108行目の上記記述でも商品のカテゴリー名が表示されるようにしましょう！

tatsuki-kato reviewed (commented):
レビュー依頼ありがとうございます！
コメントしましたので、確認をお願いします！
またお手数ですが、次回レビュー依頼の際に変更後の以下のキャプチャ動画を添付いただけないでしょうか？
```
ログアウト状態のユーザーが、商品詳細ページへ遷移した動画
```

お手数おかけしますが、よろしくお願いします！

takanori-hagiwara committed:
投稿詳細機能一覧 修正

ログアウト状態のユーザーが、商品詳細ページへ遷移した動画
https://gyazo.com/6a3a4ccb48430cfe0b47339e2edf94cf

ky3090 commented on app/views/items/show.html.erb:
```diff
@@ -103,7 +105,7 @@
   <%# 詳細ページで表示されている商品のカテゴリー名を表示しましょう %>
-  <a href="#" class="another-item"><%= "商品のカテゴリー名" %>をもっと見る</a>
+  <a href="#" class="another-item"><%= @item.category.name %>をもっと見る</a>
   <%# //詳細ページで表示されている商品のカテゴリー名を表示しましょう %>
```
```
<%# //詳細ページで表示されている商品のカテゴリー名を表示しましょう %>
```

107,109行目のコメントアウトは実装説明のコメントアウトですので、削除しましょう。

ky3090 reviewed (commented):
レビュー依頼ありがとうございます。
修正点確認させていただきました。
追加でコメントいたしましたので、ご確認お願いいたします。

takanori-hagiwara committed:
商品詳細表示機能　修正2

haramiyo commented:
レビュー依頼ありがとうございます！
こちらLGTMとさせていただきます！

実装お疲れ様でした！
プルリクエストのマージと、ローカルへのプルを忘れずに次の実装へ進んでください！
引き続き実装頑張りましょう！
